src/Extensions/Script/Serialization/JavaScriptSerializer.cs

```diff
@@ -1,42 +1,257 @@
 // MIT License.
 
-using System.Collections.Specialized;
+using System.Diagnostics.CodeAnalysis;
 using System.Text;
+using System.Text.Json;
+using System.Web.Resources;
 
 namespace System.Web.Script.Serialization;
 
-// TODO: We'll want to use System.Text.Json and change it to be strongly typed
-internal sealed class JavaScriptSerializer
+public class JavaScriptSerializer
 {
-    public object MaxJsonLength { get; internal set; }
+    internal const string ServerTypeFieldName = "__type";
+    internal const int DefaultRecursionLimit = 100;
+    internal const int DefaultMaxJsonLength = 2097152;
+    internal JsonSerializerOptions _serializerOptions;
 
-    internal static string SerializeInternal(OrderedDictionary orderedDictionary)
+    internal static string SerializeInternal(object o)
     {
-        throw new NotImplementedException();
+        JavaScriptSerializer serializer = new JavaScriptSerializer();
+        return serializer.Serialize(o);
     }
 
-    internal void Serialize(object value, StringBuilder builder, SerializationFormat javaScript)
+    internal static object Deserialize(JavaScriptSerializer serializer, string input, Type type, int depthLimit)
     {
-        throw new NotImplementedException();
+        ArgumentNullException.ThrowIfNull(input);
+
+        if (input.Length > serializer.MaxJsonLength)
+        {
+            throw new ArgumentException(AtlasWeb.JSON_MaxJsonLengthExceeded, nameof(input));
+        }
+
+        // run backward compatibility logic only if someone has invoked registered converters else use default
+        // If we can do better let me know.
+        if (serializer.Converters != null && serializer.Converters.Count > 0)
+        {
+            //backward compatibility
+            serializer.SerializerOptions.Converters.Add(new CustomDeserializerJsonConverter());
```

Comment:
this will add the converter each time you serialize something if there is a converter. Instead of tracking JavaScriptConverter, when someone registers one, just wrap it as a JsonConverter. If we need to be able to get at the JavaScriptConverter instances, that can be exposed on the custom JsonConverter.

Reply:
Taylor, as per my understanding , we shouldn't cache the serializer option, the only thing that we are using which is common across serialize/deserialize is maxdepth, otherwise Conveters (STJ) are different both serialization and de-serialization. 

Also while the Converter name is same, so confusing between JSSerializer and STJ, both have different API. If I wrap the JS Converter inside STJ and keep API and re-use serializeroptions, the same STJ conveter would be invoked during de-serialization, causing exception. LMK what you think. We can always refactor if you see improvements later, but at this point I actually don't see benefit.

Reply:
this is done.. so resolving